cvapr_data.py

```diff
@@ -67,20 +83,89 @@ def _load_evals(eval_path):
             eval_list.append( ( int(line_parts[-2][0]), int(line_parts[-1][0]) ) )
     return eval_list
 
+
+class PictureBlockData:
+
+    '''Represents data gathered during one picture block'''
+
+    def __init__(self, single_eval, block_eeg, low_freq = None, high_freq = None):
+        '''Constructs PictureBlockData object based on data gathered from a picture block
+
+        Parameters:
+            - single_eval - tuple of patient's arousal and valence evaluation
+                (or may be other way around, it's just the way they are ordered in file)
+            - block_eeg - EEG gathered during the picture block, provided by mne library
+            - low_freq - low frequency at which the signal was filtered, in Hz
+                or None in case of no filtering
+            - high_freq - low frequency at which the signal was filtered, in Hz
+                or None in case of no filtering
+        '''
+        self._low_freq = low_freq
+        self._high_freq = high_freq
+        self._mne_eeg = block_eeg
+        self._evaluation = single_eval
+
+    @property
+    def mne_eeg(self):
+        '''Wrapped mne EEG object'''
+        return self._mne_eeg
+
+    @property
+    def evaluation(self):
+        '''Tuple of patient's arousal and valence evaluation
+        (or may be other way around, it's just the way they are ordered in file)'''
+        return self._evaluation
+
+    @property
+    def raw_eeg(self):
+        '''Samples of patient's EEG,
+        It is a np.array, with first dimension being channel number, and second - sample number'''
+        return self.mne_eeg[:,:][0]
+
+    def power_spectrum(self, low_freq, high_freq, step = 1):
+        '''Returns a tuple,
+        where first element is a np.array
+        where i-th element is the power spectrum data for i-th frequency
+        and second element is a np.array
+        where i-th element is the i-th frequency, in Hz.
+        
+        The frequencies will be from low_freq to high_freq, inclusive, with step step.'''
+        if low_freq is None:
+            low_freq = self._low_freq
+        if high_freq is None:
+            high_freq = self._high_freq
+        sample_freq = self.mne_eeg.info["sfreq"]
+        return psd.psd_welch(self.mne_eeg, self._low_freq, self._high_freq, n_fft = round(sample_freq / step))

```

Comment:
I believe it should be (self.mne_eeg, **low_freq, high_freq**...). Otherwise we can't control these parameters through function arguments. Also with this line as is, an exception is thrown when using this function.

Reply:
Fixed this and some other things i've missed previously due to this unnoticed error.